modules/i3s/src/lib/parsers/parse-slpk/index.ts

```diff
@@ -0,0 +1,54 @@
+import {SlpkLoaderOptions} from 'modules/i3s/src/i3s-slpk-loader';
+import CDFileHeader from './cd-file-header';
+import LocalFileHeader from './local-file-header';
+import SlpkArchieve from './slpk-archieve';
+
+const getByteAt = (offset: number, buffer: DataView) => {
+  return buffer.getUint8(buffer.byteOffset + offset);
+};
+
+export async function parceSlpk(data: ArrayBuffer, options: SlpkLoaderOptions = {}) {
+  const slpkArchieve = new DataView(data);
+
+  const searchWindow = [
+    getByteAt(slpkArchieve.byteLength - 1, slpkArchieve),
+    getByteAt(slpkArchieve.byteLength - 2, slpkArchieve),
+    getByteAt(slpkArchieve.byteLength - 3, slpkArchieve),
+    undefined
+  ];
+
+  let hashCDOffset = 0;
+
+  for (let i = slpkArchieve.byteLength - 4; i > -1; i--) {
+    // looking for the last record in the central directory
+    searchWindow[3] = searchWindow[2];
+    searchWindow[2] = searchWindow[1];
+    searchWindow[1] = searchWindow[0];
+    searchWindow[0] = getByteAt(i, slpkArchieve);
+    if (searchWindow.toString().includes('80,75,1,2')) {
```

Comment:
I don't like serialisation. It could diminish performance.
'80,75,1,2' - add some comment for this magic sequence with link to the spec